src/main/kotlin/com/londogard/textgen/languagemodels/LanguageModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.londogard.textgen.languagemodels
+
+import com.londogard.textgen.NGram
+import smile.nlp.tokenizer.SimpleTokenizer
+import smile.nlp.tokenizer.Tokenizer
+import kotlin.system.measureTimeMillis
+
+internal typealias InternalMutableLanguageModel = HashMap<List<Int>, MutableMap<Int, Double>>
+typealias InternalLanguageModel = Map<List<Int>, Map<Int, Double>>
+
+class LanguageModel2(private val tokenizer: Tokenizer = SimpleTokenizer(), val n: Int) {
+    private val dictionary: HashMap<Int, String> = hashMapOf()
+    private val internalLanguageModel: InternalMutableLanguageModel = hashMapOf()
+    private val defaultValue = 0.0
```

Comment:
Value is only used once atm. Maybe unnecessary to have a val for it?

Reply:
Yeah, let's see.